site/hooks/useWeb3/reducers.tsx

```diff
@@ -0,0 +1,68 @@
+import { ethers } from "ethers";
+
+export type Web3ProviderState = {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  provider: any;
+  web3Provider: ethers.providers.Web3Provider | null | undefined;
+  address: string | null | undefined;
+  network: ethers.providers.Network | null | undefined;
+  connect?: () => Promise<void>;
+  disconnect?: () => Promise<void>;
```

Comment:
Thanks for linking to the Blog Post.
Just stumbled over `?` and `undefined` and thought that these values can never be undefined as soon as the wallet is connected.

Reply:
Good pickup, I've simplified the type declarations a bit :)